Lobby/EOSLobby.cs

```diff
@@ -0,0 +1,197 @@
+        }
+        
+        //find lobbies
+        search.Find(new LobbySearchFindOptions { LocalUserId = EOSSDKComponent.LocalUserProductId }, null, (LobbySearchFindCallbackInfo callback) => 
+        {
+            List<LobbyDetails> foundLobbies = new List<LobbyDetails>();
+
+            //if the search was unsuccessful, log an error and return
+            if (callback.ResultCode != Result.Success)
+            {
+                Debug.LogError("There was an error while finding lobbies. Error: " + callback.ResultCode);
+                return;
+            }
+
+            foundLobbies.Clear();
+
+            //for each lobby found, add data to details
+            for (int i = 0; i < search.GetSearchResultCount(new LobbySearchGetSearchResultCountOptions { }); i++)
+            {
+                LobbyDetails lobbyInformation;
+                search.CopySearchResultByIndex(new LobbySearchCopySearchResultByIndexOptions { LobbyIndex = (uint)i }, out lobbyInformation);
+                foundLobbies.Add(lobbyInformation);
+            }
+
+            //invoke event
+            FindLobbiesComplete.Invoke(foundLobbies);
+        });
+    }
+
+    //join lobby
+    public virtual void JoinLobby(LobbyDetails lobbyToJoin, string[] attributeKeys, bool presenceEnabled = false)
+    {
+        //join lobby
+        EOSSDKComponent.GetLobbyInterface().JoinLobby(new JoinLobbyOptions { LobbyDetailsHandle = lobbyToJoin, LocalUserId = EOSSDKComponent.LocalUserProductId, PresenceEnabled = presenceEnabled }, null, (JoinLobbyCallbackInfo callback) => 
+        {
+            List<Attribute> lobbyData = new List<Attribute>();
+
+            //if the result was not a success, log error and return
+            if(callback.ResultCode != Result.Success)
+            {
+                Debug.LogError("There was an error while joining a lobby. Error: " + callback.ResultCode);
+                return;
+            }
+
+            Attribute hostAddress = new Attribute();
+            lobbyToJoin.CopyAttributeByKey(new LobbyDetailsCopyAttributeByKeyOptions { AttrKey = hostAddressKey }, out hostAddress);
+            lobbyData.Add(hostAddress);
+
+            foreach (string key in attributeKeys)
+            {
+                Attribute attribute = new Attribute();
+                lobbyToJoin.CopyAttributeByKey(new LobbyDetailsCopyAttributeByKeyOptions { AttrKey = key }, out attribute);
+                lobbyData.Add(attribute);
+            }
+
+            isLobbyOwner = false;
+            ConnectedToLobby = true;
+            EOSSDKComponent.GetLobbyInterface().CopyLobbyDetailsHandle(new CopyLobbyDetailsHandleOptions { LobbyId = callback.LobbyId, LocalUserId = EOSSDKComponent.LocalUserProductId }, out ConnectedLobbyDetails);
+            currentLobbyId = callback.LobbyId;
+
+            //invoke event
+            JoinLobbyComplete.Invoke(lobbyData);
+        });
+    }
+
+    //leave lobby
+    public virtual void LeaveLobby()
```

Comment:
We need to hook into the eos lobby update events and listen to lobby changes (lobby modification/destruction) and then invoke events for that too. The client/server needs to be notified if the other one disconnected and there should also be user joined lobby events